### DISCUSSION TITLE
How to make personal customization work?

### DISCUSSION THREAD

**Discussion by LYK-love** [ayamir/nvimdots#819] [Q&A] [answered]
To support Debug for Rust I change the content of `dap.configurations.rust`. By default it works the same as `dap.configurations.C`:
```
	dap.configurations.c = {
		{
			name = "Launch",
			type = "lldb",
			request = "launch",
			program = utils.input_exec_path(),
			cwd = "${workspaceFolder}",
			args = utils.input_args(),
			env = utils.get_env(),

			-- if you change `runInTerminal` to true, you might need to change the yama/ptrace_scope setting:
			--
			--    echo 0 | sudo tee /proc/sys/kernel/yama/ptrace_scope
			--
			-- Otherwise you might get the following error:
			--
			--    Error on launch: Failed to attach to the target process
			--
			-- But you should be aware of the implications:
			-- https://www.kernel.org/doc/html/latest/admin-guide/LSM/Yama.html
			runInTerminal = false,
		},
	}
```

I modified the config for rust as such:
```
	dap.configurations.rust = {
		{
		  type = "rust",
		  request = "launch",
		  name = "lldb",
		  program = function()
			local metadata_json = vim.fn.system "cargo metadata --format-version 1 --no-deps"
			local metadata = vim.fn.json_decode(metadata_json)
			local target_name = metadata.packages[1].targets[1].name
			local target_dir = metadata.target_directory
			return target_dir .. "/debug/" .. target_name
		  end,
		  args = utils.get_env(),
		  env = utils.get_env(),
		  runInTerminal = false,

		},
	  }
```

Then i close the nvim and the terminal emulator. When i restart nvim and use `F8` to toggle breakpoint and `F6` to start debugging. It shows that my cusomization to `dap.configurations.rust` didn't work, the config behaves still the same as `dap.configurations.c`.

This problem is not limited to `dap.configurations.rust`, when I cusomtize my keyboard mapping in `lua/core/mapping.lua`. My config still didn't work too.
Hoewever, some config change can work, such as the change of the head image of dashboard( in `lua/modules/configs/ui/alpha.lua`). 

I consider it as `lazy-nvim`'s problem initially. But I sync all the plugins and even delete the plugins folder:
```
rm -r ~/.local/share/nvim/site/lazy/
```
The problem still exists.

So,  I wonder when I change the lua files, how to make it work?

**Comment by ayamir** [comment, marked as answer]
1. You should change `settings.lua` first before you want to enable `lldb`.
https://github.com/ayamir/nvimdots/blob/f1125a96acc769a21a7443ae1812dfdfe81812de/lua/core/settings.lua#L113
2. What keymap did you set? I guess it was caused by you set a keymap which has been overrideed to map to another command. Because we load `mapping.lua` first and `keymap` second.
https://github.com/ayamir/nvimdots/blob/f1125a96acc769a21a7443ae1812dfdfe81812de/lua/core/init.lua#L156-L157
Check the keymap you want to set whether has been set or not: `:verbose [i/n/v]map <your-keymap>`
